File: 2023-2024/sourcecode/mod4/lab12/ResourcePatternBasicsJava/app/src/main/java/com/example/resourcepatternbasicsjava/ui/CommentViewHolder.java
package com.example.resourcepatternbasicsjava.ui;

import androidx.recyclerview.widget.RecyclerView;

import com.example.resourcepatternbasicsjava.data.CommentResponseItem;
import com.example.resourcepatternbasicsjava.databinding.RvItemBinding;

public class CommentViewHolder extends RecyclerView.ViewHolder {
    private final RvItemBinding binding;

    public CommentViewHolder(RvItemBinding binding) {
        super(binding.getRoot());
        this.binding = binding;
    }

    public void bind(CommentResponseItem item) {
        binding.title.setText(item.getName());
        binding.body.setText(item.getBody());
        binding.commentId.setText(String.valueOf(item.getPostId()));
    }
}

File: 2023-2024/sourcecode/mod4/lab12/ResourcePatternBasicsKotlin/app/src/main/java/com/example/resourcepatternbasicskotlin/ui/CommentViewHolder.kt
package com.example.resourcepatternbasicskotlin.ui

import androidx.recyclerview.widget.RecyclerView
import com.example.resourcepatternbasicskotlin.data.CommentResponseItem
import com.example.resourcepatternbasicskotlin.databinding.RvItemBinding

class CommentViewHolder(private val binding: RvItemBinding) : RecyclerView.ViewHolder(binding.root) {
    fun bind(item: CommentResponseItem) {
        binding.apply {
            title.text = item.name
            body.text = item.body
            commentId.text = item.postId.toString()
        }
    }
}